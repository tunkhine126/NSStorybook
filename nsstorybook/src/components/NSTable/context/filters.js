import React, { useState, createContext } from 'react';
import PropTypes from 'prop-types';

export const NSFiltersContext = createContext();

export const NSFiltersProvider = ({
  children,
  filterCategories,
  lastUpdateProps,
  dateRangeProps,
  handleSelected,
}) => {
  const types = {
    multi: {},
    levels: {},
    single: {},
    date: {},
  };

  const { lastUpdate } = lastUpdateProps;

  ['multi', 'levels', 'single', 'date'].forEach(type => {
    filterCategories
      .filter(c => c.type === type)
      .forEach(({ id }) => {
        types[type][id] = [];
      });
  });

  const [singleSelectOptions, setSingleSelectOptions] = useState(types.single);
  const [multiSelectOptions, setMultiSelectOptions] = useState(types.multi);
  const [levelSelectOptions, setLevelSelectOptions] = useState(types.levels);
  const [showCalendar, setShowCalendar] = useState(
    lastUpdate.query === 'custom'
  );
  const [localDateRange, setlocalDateRange] = useState({
    before: null,
    after: null,
  });

  return (
    <NSFiltersContext.Provider
      value={{
        multiSelect: { multiSelectOptions, setMultiSelectOptions },
        levelSelect: { levelSelectOptions, setLevelSelectOptions },
        singleSelect: { singleSelectOptions, setSingleSelectOptions },
        calendarProps: { showCalendar, setShowCalendar },
        localDateProps: { localDateRange, setlocalDateRange },
        lastUpdateProps,
        dateRangeProps,
        filterCategories,
        handleSelected,
      }}
    >
      {children}
    </NSFiltersContext.Provider>
  );
};

NSFiltersProvider.propTypes = {
  children: PropTypes.any.isRequired,
  dateRangeProps: PropTypes.object.isRequired,
  lastUpdateProps: PropTypes.object.isRequired,
  filterCategories: PropTypes.array.isRequired,
  handleSelected: PropTypes.func.isRequired,
};
