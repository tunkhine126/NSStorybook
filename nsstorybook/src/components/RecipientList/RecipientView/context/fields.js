import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

export const FieldsContext = createContext();

export const FieldsProvider = ({ children, ...rest }) => {
  const [activePhotoField, setActivePhotoField] = useState(null);
  const [recipientPhoto, setRecipientPhoto] = useState({});

  return (
    <FieldsContext.Provider
      value={{
        activePhotoProps: { activePhotoField, setActivePhotoField },
        recipientPhotoProps: { recipientPhoto, setRecipientPhoto },
        rest,
      }}
    >
      {children}
    </FieldsContext.Provider>
  );
};

FieldsProvider.propTypes = {
  children: PropTypes.any.isRequired,
};
