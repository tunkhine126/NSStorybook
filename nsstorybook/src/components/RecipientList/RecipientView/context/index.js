import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

export const RecipientViewContext = createContext();

export const RecipientViewProvider = ({
  children,
  tabChange,
  rdData,
  data,
  profileDetails,
  primaryRecipient,
  match,
  ...rest
}) => {
  const [toaster, setToaster] = useState({
    show: false,
    recipientUuid: null,
    field: null,
    undoHandler: null,
  });
  const [saveState, setSaveState] = useState({});

  return (
    <RecipientViewContext.Provider
      value={{
        data,
        rdData,
        tabChange,
        profileDetails,
        primaryRecipient,
        match,
        toasterProps: { toaster, setToaster },
        saveProps: { saveState, setSaveState },
        rest,
      }}
    >
      {children}
    </RecipientViewContext.Provider>
  );
};

RecipientViewProvider.propTypes = {
  children: PropTypes.any.isRequired,
  tabChange: PropTypes.func.isRequired,
  profileDetails: PropTypes.object.isRequired,
  primaryRecipient: PropTypes.bool.isRequired,
  data: PropTypes.object.isRequired,
  rdData: PropTypes.any.isRequired,
  match: PropTypes.any.isRequired,
};
