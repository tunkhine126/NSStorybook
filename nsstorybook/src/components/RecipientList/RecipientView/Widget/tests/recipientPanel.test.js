/**
 *
 * Tests for data
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import TestingComponent from 'components/shared/Testing';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import RecipientPanel from '../recipientPanel';
import { RecipientViewProvider } from '../../context';

describe('<RecipientPanel />', () => {
  const props = {
    profileDetails: {
      recipientName: 'Country',
      childRecipients: 'Communities',
    },
    data: {
      name: 'Test',
      avatarUrl: null,
      internalId: '3erf23-r323d-gewf3-233',
      children: [],
      customFields: {
        uuid: '1c91d190-f5c8-11e9-a1ad-22000be1c01d',
        name: 'Jack Fresh',
        wake_up_time: null,
      },
    },
    match: {
      isExact: true,
      params: { id: '07c538ec-f5c8-11e9-a1ad-22000be1c01d' },
      path: '/families/family-list/family/:id',
      url: '/families/family-list/family/07c538ec-f5c8-11e9-a1ad-22000be1c01d',
    },
    rdData: {},
    tabChange: () => {},
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <RecipientViewProvider {...props} primaryRecipient>
          <RecipientPanel
            name="Jamie"
            type="Father"
            data={{}}
            customFields={[]}
          />
        </RecipientViewProvider>
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <RecipientViewProvider {...props} primaryRecipient={false}>
          <RecipientPanel
            name="Jamie"
            type="Father"
            data={{}}
            customFields={[]}
          />
        </RecipientViewProvider>
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
