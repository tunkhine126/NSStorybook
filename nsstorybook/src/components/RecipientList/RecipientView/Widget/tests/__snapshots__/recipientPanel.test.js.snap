// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipientPanel /> Should render and match the snapshot 1`] = `
<div
  class="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    aria-controls="family-member"
    aria-disabled="false"
    aria-expanded="false"
    class="MuiButtonBase-root MuiExpansionPanelSummary-root"
    id="jamie"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiExpansionPanelSummary-content"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
      >
        <p
          class="MuiTypography-root makeStyles-contentBody-282 MuiTypography-body2"
        >
          Jamie
        </p>
        <span
          class="MuiTypography-root makeStyles-caption-280 MuiTypography-caption"
        >
          Father
        </span>
      </div>
    </div>
    <div
      aria-disabled="false"
      aria-hidden="true"
      class="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          fill="none"
          height="15"
          viewBox="0 0 8 15"
          width="8"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M1.26658 14.9922C1.322 14.988 1.33611 14.9885 1.39097 14.9782C1.51436 14.955 1.63302 14.9039 1.73772 14.8289C1.79066 14.791 1.83687 14.7469 1.88304 14.6999L7.37893 8.69962C7.41685 8.65527 7.42743 8.64517 7.46109 8.59671C7.49887 8.54231 7.53153 8.48366 7.55841 8.42196C7.67584 8.15239 7.67584 7.83253 7.55841 7.56297C7.53153 7.50127 7.49887 7.44261 7.46109 7.38821C7.42743 7.33981 7.41685 7.32966 7.37893 7.28536L1.88304 1.2851C1.84242 1.2437 1.83312 1.2321 1.78879 1.1954C1.67236 1.09894 1.53497 1.03384 1.39097 1.00669C1.28816 0.987387 1.1825 0.987387 1.07968 1.00669C0.997469 1.02219 0.917275 1.04999 0.841936 1.08909C0.555783 1.23765 0.356557 1.54581 0.324131 1.88993C0.311353 2.02593 0.324407 2.16444 0.362282 2.29474C0.393838 2.4033 0.44243 2.50575 0.50545 2.59646C0.539112 2.64491 0.549692 2.65501 0.587614 2.69936L5.43581 7.99249L0.587614 13.2856L0.544929 13.3355C0.509892 13.3825 0.499908 13.3935 0.469406 13.4443C0.412203 13.5395 0.370068 13.6453 0.345382 13.7559C0.325643 13.8444 0.317033 13.9358 0.319872 14.0268C0.330681 14.3726 0.510213 14.6949 0.786382 14.8644C0.877293 14.9202 0.976905 14.9589 1.07968 14.9782C1.13455 14.9885 1.14865 14.988 1.20407 14.9922C1.22491 14.9922 1.24575 14.9922 1.26658 14.9922Z"
            fill="#9B9EA6"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>
  <div
    class="MuiCollapse-container MuiCollapse-hidden"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner"
      >
        <div
          aria-labelledby="jamie"
          id="family-member"
          role="region"
        >
          <div
            class="styles__ExpansionPanelDetailsStyle-qa8k4k-5 bloNLa"
          >
            <div
              class="MuiExpansionPanelDetails-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                >
                  <img
                    class="MuiCardMedia-root makeStyles-recipientChildPhoto-315 MuiCardMedia-media MuiCardMedia-img"
                    src="IMAGE_MOCK"
                    width="100%"
                  />
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <hr
                    class="MuiDivider-root MuiDivider-light"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
